var/spack/repos/builtin/packages/mumps/package.py

```diff
@@ -164,10 +164,36 @@ def install(self, spec, prefix):
 
         install_tree('lib', prefix.lib)
         install_tree('include', prefix.include)
+
+        # Create separate libexec directory tree for libraries and
+        # headers that includes a "fake MPI" header needed for
+        # sequential MUMPS builds (i.e., without MPI) when linking
+        # MUMPS to some libraries (e.g., IPOPT). This "fake MPI"
+        # header will shadow headers in MPI implementations (e.g.,
+        # MPICH, OpenMPI), so there should two separate include
+        # directories: (1) without the "fake MPI" header
```

Comment:
To avoid the confusion i would probably put `libexec` related mods under `~mpi` case, until there is a need for those in `+mpi` build. 

@nrichart may have more ideas.